Snippets/MsmqTransport/MsmqTransport_All/QueueCreation/QueueCreationUtils.cs

```diff
@@ -12,21 +12,16 @@ public static class QueueCreationUtils
         public static void CreateQueue(string queueName, string account)
         {
             var path = $@"{Environment.MachineName}\private$\{queueName}";
-            if (MessageQueue.Exists(path))
+            if (!MessageQueue.Exists(path))
             {
-                using (var messageQueue = new MessageQueue(path))
+                using (var messageQueue = MessageQueue.Create(path, true))
                 {
-                    SetPermissionsForQueue(messageQueue, account);
-                    return;
+                    SetDefaultPermissionsForQueue(messageQueue, account);
                 }
             }
-            using (var messageQueue = MessageQueue.Create(path, true))
-            {
-                SetPermissionsForQueue(messageQueue, account);
-            }
         }
 
-        static void SetPermissionsForQueue(MessageQueue queue, string account)
+        static void SetDefaultPermissionsForQueue(MessageQueue queue, string account)
```

Comment:
My goal was to make it consistent with code in the core. Upon closer inspection it seems that currently 5.2.21 and 6.0.4 use grant permissions to anonymous/everyone and local admins. 6.1 only grants permissions to local admin.

I tested the `System.Messaging` API and it does not grant any permissions to the local admins group but it grants full permissions for the caller account.

I will remove everyone/anonymous as they seem to be redundant.